### PR TITLE
Refactor: Subscriptions filtering

### PR DESCRIPTION
**Pull request by darrellwarde**
[workerB](https://workerb.linearb.io/v2/badge/collaboration-page?magicLinkId=t0EZpY1)

# Description

> **Note**
> 
> Please provide a description of the work completed in this PR below

Mainly a refactor of the monolith `compare-properties.ts`, this PR breaks out all of the various exported functions in there into separate files, and moves out utils and types to their respective homes also. Doing this refactor ahead of doing authorization for Subscriptions.

## Complexity

> **Note**
>
> Please provide an estimated complexity of this PR of either Low, Medium or High

Complexity: Low

### CODE REVIEW

**Comment by neo4j-team-graphql** [comment]
Made some changes to the Neo4j GraphQL Toolbox? Or made some changes to the core library perhaps?

Check out your changes at https://neo4j-graphql-toolbox-3397.surge.sh

Use for instance a local dbms hosted on [Neo4j Desktop](https://neo4j.com/download/) or a cloud-hosted [Aura Free](https://neo4j.com/cloud/aura-free/) dbms.

**Comment by neo4j-team-graphql** [comment]
# Performance Report

No Performance Changes


<details>
    <summary>Show Full Table</summary>
    
| name | dbHits | old dbHits | time (ms) | old time (ms) | maxRows |
| ---- | ------ | ---------- | --------- | ------------- | ------- |
|  aggregations.TopLevelAggregate | 3403 | 3403 | 35 | 43 | 1134 |
|  aggregations.NestedAggregation | 16553 | 16553 | 70 | 98 | 2174 |
|  aggregations.AggregationWithWhere | 11979 | 11979 | 54 | 80 | 2174 |
|  aggregations.AggregationWhereWithinNestedRelationships | 22116987 | 22116987 | 3220 | 3449 | 2008534 |
|  aggregations.AggregationWhereWithinNestedConnections | 22116987 | 22116987 | 3111 | 2992 | 2008534 |
|  aggregations.NestedCountFromMovieToActors | 9734 | 9734 | 52 | 51 | 2174 |
|  aggregations.NestedCountFromActorsToMovie | 9937 | 9937 | 47 | 70 | 2174 |
|  aggregations.DeeplyNestedCount | 13070329 | 13070331 | 4944 | 5112 | 2008534 |
|  batch-create.BatchCreate | 4200 | 4200 | 124 | 153 | 600 |
|  batch-create.BatchCreateSmall | 77 | 77 | 57 | 82 | 11 |
|  connect.createAndConnect | 14424 | 14424 | 194 | 197 | 3003 |
|  connections.Connection | 14082 | 14082 | 64 | 75 | 2174 |
|  connections.NestedConnection | 45466 | 45468 | 105 | 151 | 4516 |
|  connections.ConnectionWithSort | 3284 | 3284 | 134 | 106 | 1040 |
|  connections.ConnectionWithSortAndCypher | 3284 | 3284 | 371 | 350 | 1040 |
|  create.SimpleMutation | 7 | 7 | 57 | 71 | 1 |
|  cypher-directive.TopLevelCypherDirective | 122100 | 122100 | 623 | 659 | 12241 |
|  cypher-directive.TopLevelCypherDirectiveWithColumnName | 153619 | 153617 | 188 | 239 | 12241 |
|  delete.SimpleDelete | 19401 | 19401 | 234 | 380 | 1040 |
|  delete.NestedDeleteInUpdate | 23202 | 23202 | 269 | 229 | 2040 |
|  2925.SingleRelationshipFilter | 6468 | 6468 | 77 | 49 | 1040 |
|  2925.NestedSingleRelationshipFilter | 22900 | 22900 | 110 | 126 | 2174 |
|  2925.SingleRelationshipRequiredFilter | 5201 | 5201 | 51 | 44 | 1040 |
|  2925.NestedSingleRelationshipRequiredFilter | 9361 | 9361 | 62 | 67 | 1040 |
|  query.SimpleQuery | 3121 | 3121 | 24 | 26 | 1040 |
|  query.SimpleQueryWithRelationship | 16162 | 16162 | 47 | 51 | 2174 |
|  query.QueryWhere | 9703 | 9705 | 39 | 60 | 2163 |
|  query.SimpleQueryWithNestedWhere | 9881 | 9883 | 54 | 70 | 2163 |
|  query.Nested | 10096041 | 10096041 | 10996 | 10835 | 2008534 |
|  query.NestedWithFilter | 10074401 | 10074401 | 10524 | 10195 | 2004000 |
|  query.OrFilterOnRelationships | 43647 | 41041 | 195 | 235 | 1946 |
|  query.OrFilterOnRelationshipsAndNested | 37520 | 34917 | 207 | 248 | 1946 |
|  query.QueryWithNestedIn | 13214 | 13363 | 57 | 62 | 1154 |
|  query.NestedConnectionWhere | 9834 | 9834 | 57 | 95 | 2174 |
|  query.DeeplyNestedConnectionWhere | 9907 | 10074 | 78 | 207 | 2174 |
|  query.DeeplyNestedWithRelationshipFilters | 19128 | 19332 | 149 | 252 | 1592 |
|  query.NestedWithRelationshipSingleFilters | 3881 | 3881 | 165 | 316 | 1134 |
|  query.Fulltext | 80 | 80 | 42 | 36 | 16 |
|  query.FulltextWithNestedQuery | 587 | 587 | 52 | 55 | 84 |
|  sorting.SortMultipleTypes | 3515 | 3515 | 92 | 89 | 1040 |
|  sorting.SortMultipleTypesWithCypherWithCypher | 1441 | 1545 | 312 | 286 | 1040 |
|  sorting.SortOnNestedFields | 14082 | 14082 | 67 | 71 | 2174 |
|  sorting.SortDeeplyNestedFields | 43198 | 43198 | 116 | 112 | 4516 |
|  sorting.SortWithTopLevelCypher | 3121 | 3121 | 122 | 203 | 1040 |
|  unions.SimpleUnionQuery | 321 | 321 | 63 | 68 | 35 |
|  unions.SimpleUnionQueryWithMissingFields | 293 | 293 | 73 | 80 | 35 |
|  unions.NestedUnion | 410637 | 410637 | 294 | 303 | 33033 |
|  unions.NestedUnionWithMissingFields | 384611 | 384611 | 285 | 359 | 33033 |
|  update.NestedUpdate | 16143 | 16143 | 121 | 213 | 2002 |

</details>
        
Old Schema Generation: 46.676s
Schema Generation: 47.251s
Old Subgraph Schema Generation: 1:01.286 (m:ss.mmm)
Subgraph Schema Generation: 1:01.796 (m:ss.mmm)

**Review by a-alle — APPROVED** [review]
Left a few nitpicks, glad to see this code refactored!

File: packages/graphql/src/schema/resolvers/subscriptions/where/utils/get-filtering-fn.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        return filtered.some((v) => v === received);
+    },
+    NOT_IN: (received: string | number, filtered: [string | number]) => {
+        return !filtered.some((v) => v === received);
+    },
+};
+
+export function getFilteringFn<T>(
+    operator: string | undefined,
+    overrides?: Record<string, (received: any, filtered: any, fieldMeta?: any) => boolean>
```
Comment (a-alle) [inline packages/graphql/src/schema/resolvers/subscriptions/where/utils/get-filtering-fn.ts:68]: As this allows for overrides from multiple sources, is it maybe useful to add a type for the keys as not any string would work?
Also we should prefer `unknown` to `any`

Comment (darrellwarde) [reply packages/graphql/src/schema/resolvers/subscriptions/where/utils/get-filtering-fn.ts:68]: I think that's a good idea! `unknown` does not work, a whole host of typing issues when I did that. 🙃

**Pull request merged**